### PR TITLE
feat: TemplateAPI初步实现

### LINKED ISSUE #53: 实现 common-lib 的模版开发
实现新架构可支持的 common-lib 模版，通过 npm 拉取、本地引入等方式拉近项目文件。这个过程要注意同时适配 ts 这一类插件内容。
类同 vue、react 的模版实现，因为比较硬核，需要贡献者提供一个详细的实现方案，评审通过后进行开发。

### PR DESCRIPTION
**Pull request by nonhana**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/xun082/create-neat/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

- Closes #53 
- Related to #53 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/core/src/utils/ConfigTransform.ts [hunk 1]
```diff
@@ -41,7 +54,7 @@ export default class ConfigTransform {
    * @param context 上下文
    * @returns {filename, content} filename为文件名 content为转化后得文件内容
    */
-  thransform(value, files, context) {
+  transform(value, files, context) {
     const file = this.getDefaultFile();
     const { type, filename } = file;
     const transform = transforms[type];
```
Comment (qq1051766345) [inline packages/core/src/utils/ConfigTransform.ts:60]: 这里函数和变量共用一个transform我觉得不太好，或许可以改下名字让语义更清晰？

Comment (nonhana) [reply packages/core/src/utils/ConfigTransform.ts:60]: 是这样，我顺便处理一下

File: packages/@plugin/plugin-babel/index.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const buildToolConfigs = {
   // 添加其他构建工具的配置...
 };
 
-const pluginBable = (api, options) => {
+const bablePlugin = (api, options) => {
```
[the remark this answers is not in the document]
Comment (nonhana) [reply packages/@plugin/plugin-babel/index.js:15]: 好的，我把这里的部分全部重新还原一下，由对应编写插件的人负责。

File: packages/core/src/utils/ConfigTransform.ts (hunk 1 above)
Comment (qq1051766345) [reply packages/core/src/utils/ConfigTransform.ts:60]: 这个记得也一起改一下

**Pull request merged**